upgrade-prechecks/preupgrade-healthcheck.sh

```diff
@@ -919,42 +928,97 @@ function verify_pid_limit(){
+    fi
+}
+
 rm -f ${REPORT} > /dev/null
-print_header
-print_section "API access"
-verify_api_access
-print_section "Registry access"
-areRegistriesAccessible
-print_section "Registry authentication and authorisation"
-isAuthCorrect
-print_section "Cluster operators"
-verify_clusteroperators_status
-print_section "Nodes"
-verify_nodes_status
-print_section "Machine configuration pools"
-verify_mcp
-print_section "Catalog sources"
-verify_catsrc
-print_section "Fusion software"
-verify_fusion_health
-print_section "Backup & Restore"
-verify_br_health
-print_section "Legacy SPP"
-verify_spp_health
-print_section "Data Classification"
-verify_dcs_health
-print_section "Scale cluster"
-verify_scale_health
-print_section "VMs migration"
-verify_livemigratable_vms
-print_section "Pods with imagepullbackoff across cluster"
-verify_imagepullbackoff_pods
-print_section "Nodes hardware status"
-verify_nodes_hw
-print_section "Fusion nodes maintenance"
-verify_node_taints
-print_section "Network checks"
-verify_network_checks
-print_section "Verify PID limit"
-verify_pid_limit
-print_footer
+
+if [ $# -eq 0 ]; then
+        print_header
+        print_section "API access"
+        verify_api_access
+        print_section "Registry access"
+        areRegistriesAccessible
+        print_section "Registry authentication and authorisation"
+        isAuthCorrect
+        print_section "Cluster operators"
+        verify_clusteroperators_status
+        print_section "Nodes"
+        verify_nodes_status
+        print_section "Machine configuration pools"
+        verify_mcp
+        print_section "Catalog sources"
+        verify_catsrc
+        print_section "Fusion software"
+        verify_fusion_health
+        print_section "Backup & Restore"
+        verify_br_health
+        print_section "Legacy SPP"
+        verify_spp_health
+        print_section "Data Classification"
+        verify_dcs_health
+        print_section "Scale cluster"
+        verify_scale_health
+        print_section "VMs migration"
+        verify_livemigratable_vms
+        print_section "Pods with imagepullbackoff across cluster"
+        verify_imagepullbackoff_pods
+        print_section "Nodes hardware status"
+        verify_nodes_hw
+        print_section "Fusion nodes maintenance"
+        verify_node_taints
+        print_section "Network checks"
+        verify_network_checks
+        print_section "Verify PID limit"
+        verify_pid_limit
+        print_section "Verify Presence of CSI Configmap"
+        verify_CSI_configmap_present
+        print_footer
+elif [ "$1" == "postcheck" ]; then
+        print_header
+        print_section "API access"
+        verify_api_access
+        print_section "verify token secret in Scale service account"
+        verify_token_secret_present
+        print_footer
+else
+        print info "For postcheck validation proper argument not provided. To invoke postchecks pass the argument 'postcheck'"
```

Comment:
print info "For postcheck validation proper argument not provided. To invoke postchecks pass the argument 'postcheck'"

this should be pronted in error format

it should say, if you want to run precheck then do not pass argument and if post check then argument "postcheck" needs to be passed

BTW, in previous comment I had suggested to include usage method.

Reply:
done.